.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "ldk"]
+	path = ldk
+	url = https://github.com/lightningdevkit/ldk-garbagecollected.git
```

Comment:
We currently use `v0.0.98.3` of LDK since all newer versions have a failing test:
```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.ldk.ManualMsgHandlingPeerTest
Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec
Running org.ldk.HumanObjectPeerTest
Running test with flags 0
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.021 sec <<< FAILURE!
org.ldk.HumanObjectPeerTest.test_message_handler()  Time elapsed: 0.02 sec  <<< FAILURE!
java.lang.ExceptionInInitializerError
	at org.ldk.structs.Logger.<init>(Logger.java:17)
	at org.ldk.structs.Logger.new_impl(Logger.java:35)
	at org.ldk.HumanObjectPeerTestInstance$Peer.<init>(HumanObjectPeerTest.java:208)
	at org.ldk.HumanObjectPeerTestInstance$Peer.<init>(HumanObjectPeerTest.java:297)
	at org.ldk.HumanObjectPeerTestInstance.do_test_message_handler(HumanObjectPeerTest.java:610)
	at org.ldk.HumanObjectPeerTest.do_test_run(HumanObjectPeerTest.java:879)
	at org.ldk.HumanObjectPeerTest.do_test(HumanObjectPeerTest.java:884)
	at org.ldk.HumanObjectPeerTest.test_message_handler(HumanObjectPeerTest.java:911)
Caused by: java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:208)
	at java.base/java.nio.file.Files.copy(Files.java:3112)
	at org.ldk.impl.bindings.<clinit>(bindings.java:32)
	... 26 more

Running org.ldk.PeerTest
Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec

Results :

Failed tests:   org.ldk.HumanObjectPeerTest.test_message_handler()

Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
```

Reply:
This implies that a matching `liblightningjni` native library was not found - how are you importing the LDK dependency and what platform are you running on?

Reply:
LDK is declared as a Maven module `ldk`:
https://github.com/getlipa/ldk-sample-java/blob/c5d2fca04c3b6d87fbb4a1247f97ae097c8b3197/pom.xml#L12

The `ldk-sample` module then decalres it as a dependency:
https://github.com/getlipa/ldk-sample-java/blob/c5d2fca04c3b6d87fbb4a1247f97ae097c8b3197/ldk-sample/pom.xml#L15-L19

Tests are run either through the parent module by `LD_LIBRARY_PATH=$PWD/ldk ./mvnw verify` or within the `ldk` module / directory directly by `LD_LIBRARY_PATH=$PWD mvn verify`. Both lead to the same test failure mentioned in the initial comment.

The platform is a `Ubuntu 20.04.2 LTS` with kernel `5.4.0-77-generic` (x86_64) and Java `11.0.11, vendor: Ubuntu, runtime: /usr/lib/jvm/java-11-openjdk-amd64`

Reply:
Is it possible the files in `resources` are being stripped by your build process? Note that the `resources` files are not included in the git tree, only in the release binaries (though you can fetch them directly from https://git.bitcoin.ninja/ldk-java-bins if you want).

Reply:
Since LDK is now on maven central, this one is no longer a real blocker. But maybe I will try to find out more anyway.